ros_ws/src/teleoperation/src/joystick_controller.cpp

```diff
@@ -14,8 +17,21 @@ JoystickController::JoystickController()
     this->m_joystick_subscriber =
         this->m_node_handle.subscribe<sensor_msgs::Joy>("joy", 10, &JoystickController::joystickCallback, this);
 
+
+    // get the provided gamepad type
     ros::NodeHandle private_node_handle("~");
-    private_node_handle.getParam(INVERT_STEERING_PARAMETER, this->m_invert_steering);
+    private_node_handle.getParam(GAMEPAD_TYPE_PARAMETER, this->m_gamepad_type);
+
+    // load joystick map for the provided gamepad type
+    ROS_ASSERT_MSG(m_gamepad_type != "xbox360" || m_gamepad_type != "ps3", "Gamepad type should be xbox360 or ps3");
+    if (m_gamepad_type == "xbox360")
```

Comment:
Is this how string comparisons are done in C++? Afaik, this shouldn't work (use something like strcmp instead). Did you test your code with both controllers?

Reply:
This type of comparison works most of the time but yes it shouldn't work. Thats why we got the (std::string literal)[https://en.cppreference.com/w/cpp/string/basic_string/operator""s] in C++11.

Reply:
Actually i am still not sure how to compare strings properly in c++

Reply:
Sorry, my fault. == is fine ^^